src/scheduler/multi_threading.cpp

```diff
@@ -237,8 +239,8 @@ worker(void *tid)
 	pthread_setspecific(th_id_key, tid);
 	ntid = *(int *)tid;
 
-	/* Block HUPs, if we ever unblock this, we'll need to modify 'restart()' to handle MT */
-	sigemptyset(&set);
+	/* Add HUP to the list of signals to block, if we ever unblock this, we'll need to modify 'restart()' to handle MT */
+	set = allsigs;
```

Comment:
Shouldn't the threads block allsigs all the time? This looks like that it will block momentarily and unblock later on. So if the scheduler receives a signal after it is out of threading code but still within a cycle, one of these threads can potentially honor the signal even though the scheduler blocked it for the entire cycle.

Reply:
they do block them all the time right? each thread will block these signals until they die, we never unblock them right? The main thread will unblock allsigs and block them again every cycle, but that will not affect the fact that the worker threads have these blocked, right?

Reply:
Oh ok, I thought worker threads unblock them later on. This looks good then.